atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/sweep/queue/TargetedSweepTest.java

```diff
@@ -313,4 +316,25 @@ public void cleanup() {}
     private void sweepNextBatch(ShardAndStrategy shardStrategy) {
         sweepQueue.sweepNextBatch(shardStrategy, Sweeper.of(shardStrategy).getSweepTimestamp(sweepTimestampSupplier));
     }
+
+    /**
+     * After this method returns successfully, we know that there has existed some point in time when the immutable
+     * timestamp from the {@link #serializableTxManager} was strictly greater than the timestamp parameter that was
+     * passed. In the absence of concurrent transaction starts, this also means that the immutable timestamp will be
+     * strictly greater than the timestamp parameter going forward.
+     *
+     * Note that in the general case, this does not guarantee that the immutable timestamp is currently greater than
+     * the timestamp parameter in the presence of concurrent transaction starts, since under certain scheduling
+     * conditions it may go backwards.
+     *
+     * The purpose of this method is to ensure that, for tests that assert that Sweep has deleted values written in a
+     * given transaction, that that transaction has released its immutable timestamp lock (and so Sweep can make
```

Comment:
double that?

Reply:
Actually this is intentional: read this as `that [that transaction] has released its immutable timestamp lock`

Reply:
i.e. that transaction is said given transaction